garrysmod/lua/includes/extensions/string.lua

```diff
@@ -96,255 +80,215 @@ function string.Explode(separator, str, withpattern)
+	if ( !string_find( path, "\\", 1, true ) && !string_find( path, "/", 1, true ) ) then return path end
+	return string_match( path, "[\\/]([^/\\]+)$" ) or ""
 end
 
---[[-----------------------------------------------------------------
-	Name: FormattedTime( TimeInSeconds, Format )
-	Desc: Given a time in seconds, returns formatted time
-			If 'Format' is not specified the function returns a table
-			conatining values for hours, mins, secs, ms
-
-	Examples: string.FormattedTime( 123.456, "%02i:%02i:%02i")	==> "02:03:45"
-			  string.FormattedTime( 123.456, "%02i:%02i")		==> "02:03"
-			  string.FormattedTime( 123.456, "%2i:%02i")		==> " 2:03"
-			  string.FormattedTime( 123.456 )					==> { h = 0, m = 2, s = 3, ms = 45 }
--------------------------------------------------------------------]]
-function string.FormattedTime( seconds, format )
-	if ( not seconds ) then seconds = 0 end
-	local hours = math.floor( seconds / 3600 )
-	local minutes = math.floor( ( seconds / 60 ) % 60 )
-	local millisecs = ( seconds - math.floor( seconds ) ) * 100
-	seconds = math.floor( seconds % 60 )
-
-	if ( format ) then
-		return string.format( format, minutes, seconds, millisecs )
-	else
-		return { h = hours, m = minutes, s = seconds, ms = millisecs }
+local function string_FormattedTime( seconds, format )
+	local minutes = math_floor( seconds / 60 % 60 )
+	local millisecs = ( seconds - math_floor( seconds ) ) * 100
+	seconds = math_floor( seconds % 60 )
+
+	if ( format == nil ) then
+		return { h = math_floor( seconds / 3600 ), m = minutes, s = seconds, ms = millisecs }
 	end
+
+	return string_format( format, minutes, seconds, millisecs )
 end
 
---[[---------------------------------------------------------
-	Name: Old time functions
------------------------------------------------------------]]
-function string.ToMinutesSecondsMilliseconds( TimeInSeconds ) return string.FormattedTime( TimeInSeconds, "%02i:%02i:%02i" ) end
-function string.ToMinutesSeconds( TimeInSeconds ) return string.FormattedTime( TimeInSeconds, "%02i:%02i" ) end
+string.FormattedTime = string_FormattedTime
 
-local function pluralizeString( str, quantity )
-	return str .. ( ( quantity ~= 1 ) and "s" or "" )
+function string.ToMinutesSecondsMilliseconds( seconds )
+	return string_FormattedTime( seconds, "%02i:%02i:%02i" )
 end
 
-function string.NiceTime( seconds )
+function string.ToMinutesSeconds( seconds )
+	return string_FormattedTime( seconds, "%02i:%02i" )
+end
 
-	if ( seconds == nil ) then return "a few seconds" end
+function string.NiceTime( seconds )
+	if ( seconds <= 0 ) then
+		return "0"
+	end
 
 	if ( seconds < 60 ) then
-		local t = math.floor( seconds )
-		return t .. pluralizeString( " second", t )
+		local t = math_floor( seconds )
+		return t .. ( t == 1 and " second" or " seconds" )
 	end
 
-	if ( seconds < 60 * 60 ) then
-		local t = math.floor( seconds / 60 )
-		return t .. pluralizeString( " minute", t )
+	-- 60 * 60
+	if ( seconds < 3600 ) then

```

Comment:
Side note: it's totally fine to write literal expressions. They are `consteval`

Reply:
Didn't know that at the time I wrote this.